### PR TITLE
Handle old-style otlp endpoints properly.

### PR DESCRIPTION
**Pull request by jkwatson**
non-scheme'd endpoints would blow up before this change.

### CODE REVIEW

File: exporters/otlp/trace/src/test/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterTest.java [hunk 1]
```diff
@@ -83,13 +87,12 @@ void invalidConfig() {
     assertThatThrownBy(() -> OtlpGrpcSpanExporter.builder().setEndpoint(null))
         .isInstanceOf(NullPointerException.class)
         .hasMessage("endpoint");
-    assertThatThrownBy(() -> OtlpGrpcSpanExporter.builder().setEndpoint("😺://localhost"))
```
Comment (jkwatson) [inline exporters/otlp/trace/src/test/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterTest.java:86]: sorry, no more kitty-cats. they'll have to fail when grpc tries to connect

File: exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java [hunk 2]
```diff
@@ -80,7 +80,11 @@ public OtlpGrpcSpanExporterBuilder setTimeout(Duration timeout) {
    */
   public OtlpGrpcSpanExporterBuilder setEndpoint(String endpoint) {
     requireNonNull(endpoint, "endpoint");
-    final URI uri;
+    if (!endpoint.matches("^[a-zA-Z]+:\\/\\/.*")) {
```
Comment (anuraaga) [inline exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java:83]: How about just `endpoint.contains("://")`?  That way we won't miss restoring invalid URL check when we remove support for schemaless URLs.

**Review by anuraaga — APPROVED** [review]
Sorry thought there was still a residual test for this but there wasn't, oops.

File: exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java (hunk 2 above)
Comment (jkwatson) [reply exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java:83]: Seems fine. I'll update

Comment (jkwatson) [reply exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java]: done

File: exporters/otlp/trace/src/test/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterTest.java (hunk 1 above)
Comment (anuraaga) [reply exporters/otlp/trace/src/test/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterTest.java:86]: Aww thought we saved the cat 😿

Comment (jkwatson) [reply exporters/otlp/trace/src/test/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterTest.java:86]: ah well.. we can bring it back if you think it's an important use-case that people might fall into.  ;)

**Pull request merged**